### PR TITLE
feat: added Permission guard

### PR DESCRIPTION
**Pull request by VictoriqueMoe**
Added Permission guard to easily allow people who use global commands to add role permissions for command execution.

## Package

@discordx/utilities

### CODE REVIEW

**Comment by VictoriqueMoe** [comment]
Prtittier is an issue because i am on Windows, doing a auto format changes all the line from `CRLF` to `LF` and i get 400 changes

**Comment by oceanroleplay** [comment]
> Prtittier is an issue because i am on Windows, doing a auto format changes all the line from `CRLF` to `LF` and i get 400 changes

there is git config to commit LF, enable that for repo and then try, it should not commit all files.

File: packages/utilities/README.md [hunk 1]
```diff
@@ -175,17 +175,66 @@ class RateLimitExample {
...
 @Guard(NotBot)
 hello({ message }: SimpleCommandMessage): void {
   message.reply("It worked!");
 }
-...
 ```
 
 This will work on both Slash and Simple commands
 
+## PermissionGuard
```
Comment (oceanroleplay) [inline packages/utilities/README.md:187]: add this in index at top

Comment (VictoriqueMoe) [reply packages/utilities/README.md:187]: can you explain?

Comment (VictoriqueMoe) [reply packages/utilities/README.md:187]: never mind, i figured out what you meant

**Review by oceanroleplay — APPROVED** [review]

**Pull request merged**